src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -416,12 +452,20 @@ protected boolean setupAndConsumeRecipeInputs(Recipe recipe, IItemHandlerModifia
         IMultipleTankHandler importFluids = getInputTank();
         IMultipleTankHandler exportFluids = getOutputTank();
 
-        if (!canVoidRecipeOutputs() &&
-                (!MetaTileEntity.addItemsToItemHandler(exportInventory, true, recipe.getAllItemOutputs(exportInventory.getSlots())) ||
-                 !MetaTileEntity.addFluidsToFluidHandler(exportFluids, true, recipe.getFluidOutputs()))) {
+        // We have already trimmed outputs and chanced outputs at this time
+        // Set the outputs here so the chanced outputs calculated for attempted merge into output bus are not recalculated later
+        this.itemOutputs = GTUtility.copyStackList(recipe.getResultItemOutputs(GTUtility.getTierByVoltage(recipeEUt), getRecipeMap()));
+        if (!metaTileEntity.canVoidRecipeItemOutputs() && !MetaTileEntity.addItemsToItemHandler(exportInventory, true, this.itemOutputs)) {
+            this.isOutputsFull = true;
+            return false;
+        }
```

Comment:
As stated above, I believe `itemOuputs` could simply be set to null here.

Reply:
Even if the merge did not fail here, it could still fail on the return statement for the `recipe.matches` call.
This means that if it passes the merge into the item handlers, but not the matches call, the item outputs would not get properly reset if it was done here.

Resetting if this method returned false is just much easier to catch all the cases in one place